compiler/rustc_parse/src/parser/generics.rs

```diff
@@ -145,6 +145,19 @@ impl<'a> Parser<'a> {
                         } else {
                             (None, Vec::new())
                         };
+
+                        if this.eat(&token::Eq) {
```

Comment:
This should probably use `check` to check for the `=`, and `lookahead(1, ..)` to check for the lifetime. This is to make sure we don't eat `=` in the case we have `<'a =` without a lifetime that follows.

Reply:
The `check_noexpect` should also fix the CI failures